pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -270,8 +253,7 @@ private TransferableBlock consumeAggregation() {
         new FunctionContext(FunctionContext.Type.AGGREGATION, functionName, arguments), true);
   }
 
-  private static AggregationFunction<?, ?> getAggFunctionForIntermediateInput(RexExpression.FunctionCall functionCall,
-      Map<Integer, Literal> literalArguments) {
+  private static AggregationFunction<?, ?> getAggFunctionForIntermediateInput(RexExpression.FunctionCall functionCall) {
```

Comment:
Should we use same way to process raw input and intermediate input since we already have all operands?
Right you we assume we don't need function for raw input which is not always true

Reply:
The only difference is that first argument type in intermediate stage function operands might be different than raw input stage.

For count(*), no argument in rawInput stage, but you need one InputRef in the intermediate stage.